web/src/integTest/java/com/microsoft/applicationinsights/azuretest/CompareTelemetryWithAzureExportTests.java

```diff
@@ -0,0 +1,342 @@
+            }
+
+            if (messages.size() < config.getNumberOfMessagesToRetrieve()){
+                Helpers.sleep(config.getSecondsToSleep() * millisecondsInSecond);
+            }
+        }
+
+        ArrayList<TelemetryItem> telemetryItems = new ArrayList<TelemetryItem>();
+        for (JSONObject json : telemetryAsJson) {
+            telemetryItems.add(getTelemetryItemFromJson(docType, json));
+        }
+
+        return telemetryItems;
+    }
+
+    /**
+     * Gets a list of blob URIs from Azure queue messages
+     * @param docType The document type to retrieve from Azure
+     * @param messages The Azure queue messages
+     * @return An ArrayList of blob URIs
+     * @throws Exception
+     */
+    private ArrayList<String> getBlobUrisFromQueueMessages(DocumentType docType, ArrayList<CloudQueueMessage> messages) throws Exception {
+        ArrayList<String> blobUris = new ArrayList<String>();
+
+        System.out.println("Extracting blob URIs of document type " + docType.toString() + " from " + messages.size() + " messages");
+        for (CloudQueueMessage message : messages) {
+            JSONObject messageContentAsJson = new JSONObject(message.getMessageContentAsString());
+            String msgDocType = messageContentAsJson.getString("DocumentType");
+
+            if(msgDocType.equals(docType.toString())) {
+                blobUris.add(messageContentAsJson.getString("BlobUri"));
+            }
+        }
+
+        System.out.println("Got " + blobUris.size() + " blob URIs with document type " + docType.toString());
+        return blobUris;
+    }
+
+    /**
+     * Converts a string to multiple JSON objects
+     * @param jString The string to convert
+     * @return ArrayList of JSON objects
+     */
+    private ArrayList<JSONObject> getJsonsFromString(String jString) throws JSONException {
+        System.out.println("Extracting JSON objects from string");
+        String[] jsonStrings = jString.split("\n");
+        ArrayList<JSONObject> jsonObjects = new ArrayList<JSONObject>();
+
+        for(String s : jsonStrings) {
+            jsonObjects.add(new JSONObject(s));
+        }
+
+        System.out.println("Got " + jsonObjects.size() + " JSON objects");
+        return jsonObjects;
+    }
+
+    /**
+     * Converts JSON object to TelemetryItem by it's document type
+     * @param docType The document type of the telemetry
+     * @param json The JSON object
+     * @return A TelemetryItem
+     * @throws Exception
+     */
+    private TelemetryItem getTelemetryItemFromJson(DocumentType docType, JSONObject json) throws Exception {
+        if (docType == DocumentType.Requests) {
+            return getRequestTelemetryItemFromJson(json);
+        }
+
+        if (docType == DocumentType.PerformanceCounters) {
+            return getPerformanceCounterTelemetryItemFromJson(json);
+        }
+
+        throw new Exception("Can't find document type: " + docType.toString());
```

Comment:
You don't need to look for the docType as you get it as a parameter. It would be better to use "Unsupported document type"